From: Pierre Joye <pierre.php@gmail.com>
Date: Sat, 4 Jun 2016 23:09:01 +0700
Subject: fix #215 gdImageFillToBorder stack-overflow when invalid color is
 used

---
 src/gd.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/gd.c b/src/gd.c
index 64f6570..b369743 100644
--- a/src/gd.c
+++ b/src/gd.c
@@ -1830,11 +1830,17 @@ BGD_DECLARE(void) gdImageFillToBorder (gdImagePtr im, int x, int y, int border,
 	int i;
 	int restoreAlphaBleding;
 
-	if (border < 0) {
+	if (border < 0 || color < 0) {
 		/* Refuse to fill to a non-solid border */
 		return;
 	}
 
+	if (!im->trueColor) {
+		if ((color > (im->colorsTotal - 1)) || (border > (im->colorsTotal - 1))) {
+			return;
+		}
+    }
+
 	leftLimit = (-1);
 
 	restoreAlphaBleding = im->alphaBlendingFlag;
